### PR TITLE
Disable all sessions

### PR DESCRIPTION
**Pull request by shernanm**
## Summary

* The endpoint is implemented to disable all active sessions for a specific user.
* Their respective tests are carried out.

## Trello Card

https://trello.com/c/wrMZDcQQ

### CODE REVIEW

File: app/controllers/user.js [hunk 1]
```diff
@@ -70,3 +79,14 @@ exports.createAdminUser = (req, res, next) => {
...
     .catch(next);
 };
+
+exports.disableAllSessions = (req, res, next) => {
+  logger.info(`disableAllSessions method start, request methods: ${req.method}, endpoint: ${req.path}.`);
+  const { email } = decodedToken(req.headers.token);
+  return updateSecret(email)
+    .then(() => {
+      logger.info(`All sessions for the user ${email} have been disabled successfully`);
+      res.status(200).send({ message: 'All sessions have been disabled' });
```
Comment (raparicio6) [inline app/controllers/user.js]: you are missing the return. also, sending status 200 is by default in Express

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/middlewares/session.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+const { findUserBy } = require('../services/users');
 
 exports.tokenValidator = (req, res, next) => {
   const { token } = req.headers;
   if (token) {
     return validateToken(token)
-      .then(() => next())
+      .then(() => {
+        const { email, iat } = decodedToken(token);
+        findUserBy({ email }, ['allowedDate']).then(({ allowedDate }) => {
```
Comment (molinajulian) [inline app/middlewares/session.js]: return here

File: app/middlewares/session.js [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+          const dateToken = new Date(iat * 1000);
```
Comment (molinajulian) [inline app/middlewares/session.js]: Date? check `moment` library

File: app/models/user.js [hunk 4]
```diff
@@ -27,6 +27,12 @@ module.exports = (sequelize, DataTypes) => {
         type: DataTypes.ENUM,
         values: ['regular', 'admin'],
         defaultValue: 'regular'
+      },
+      allowedDate: {
+        allowNull: false,
+        type: DataTypes.DATE,
+        field: 'allowed_date',
+        defaultValue: Date.now()
```
Comment (molinajulian) [inline app/models/user.js]: This value `Date.now()` its constant. Check how to use literal function for save, for example `NOW()` or `SYSDATE()`

File: migrations/migrations/20190801172008-allowed-date-column.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) =>
+    queryInterface.addColumn('users', 'allowed_date', {
+      type: Sequelize.DATE
```
Comment (molinajulian) [inline migrations/migrations/20190801172008-allowed-date-column.js]: Add here `defaultValue` and `allowNull:false`

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: app/controllers/user.js [hunk 6]
```diff
@@ -50,11 +58,14 @@ exports.createAdminUser = (req, res, next) => {
   user.role = ADMIN_ROLE;
   logger.info(`createAdminUser method start, request methods: ${req.method}, endpoint: ${req.path},
   user: ${user.firstName} ${user.lastName}`);
-  findUserBy({ email })
+  findUserBy({
```
Comment (molinajulian) [inline app/controllers/user.js]: Return here

File: app/middlewares/session.js [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
 
 exports.tokenValidator = (req, res, next) => {
   const { token } = req.headers;
   if (token) {
     return validateToken(token)
-      .then(() => next())
+      .then(() => {
+        const { email, iat } = decodedToken(token);
+        return findUserBy({ email }, ['allowedDate']).then(({ allowedDate }) => {
+          const dateToken = moment.unix(iat);
```
Comment (molinajulian) [inline app/middlewares/session.js]: Is more easy ; moment(allowedDate), moment(iat) and moment.diff(moment1,moment2)

**Review by molinajulian — CHANGES_REQUESTED** [review]

**Review by raparicio6 — APPROVED** [review]

**Comment by raparicio6** [comment]
Excellent work

**Pull request merged**